libraries/AP_Camera/AP_Camera.cpp

```diff
@@ -302,6 +308,13 @@ void AP_Camera::update()
     if (is_zero(_trigg_dist)) {
         return;
     }
+
+    const AP_AHRS &ahrs = AP::ahrs();
+    Location current_loc;
+    if (!ahrs.get_position(current_loc)) {
+        // completely ignore this failure!  AHRS will provide its best guess.
```

Comment:
I'd actually disagree with this stance. If we don't have a position then shouldn't be running update any further. (I do realize this is technically a behavior change, but I think now is also the correct time to have the conversation on this sort of PR.

Reply:
"technically"?!

No, the correct place is on a follow-up PR, not on a PR where I really, really don't want behavioural changes!

There's lots of things we can fix up in `AP_Camera`.  A quick squiz shows manual determination of relative altitude (regardless of whether home is set), 50Hz loop assumption, range issues on `_trigger_counter`, direct references to GPS where there shouldn't be, not using helper functions on location where it could, not using functions like is_positive, lack of `const`, mixed use of counters and timestamps, lack of data integrity around data shared between threads, out-of-date comments, use of multiple variables in place of an enum for what we're using for timing, incorrect use of logger singleton.

None of those should be addressed in a hopefully-NFC PR!

Reply:
See that's where I have to disagree. The place to consider and discuss short comings on the failure handling is when we are looking at it, rather then just sticking it in a if to suppress the warning. The warning is there to show a spot we are not handling well, and hiding it off in the if is bad.

I know there are problems in AP_Camera, but I'm not pushing to go address all of them. I'm pointing out that you are introducing a new one that was out of it's scope with the previous incarnation and should be fixing it rather then pushing it under the rug like this.

Reply:
I'd rather not be introducing behavioural changes.

I was planning on coming back and doing to patches on top of this PR:
 - use methods on the current_loc object to get the heights in the correct frames rather than doing it ourselves
 - returning early when the AHRS indicates it doesn't know where we are.

We don't have a PR budget we're adhering to, do we?